File: binary-search/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int ans = 100001, count = 0;
        for (int l = 0, r = 0; r < nums.length; r++) {
            count += nums[r];
            while (count >= target) {
                ans = Math.min(ans, r - l + 1);
                count -= nums[l++];
            }
        }
        return ans == 100001 ? 0 : ans;
    }
}

File: binary-search/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, ans, s = 0, 100001, 0

        for r in range(len(nums)):
            s += nums[r]

            while (s >= target):
                s -= nums[l]
                ans = min(ans, r - l + 1)
                l += 1

        return 0 if ans == 100001 else ans